app/src/main/java/com/asma/tasky/di/AppModule.kt

```diff
@@ -0,0 +1,47 @@
+package com.asma.tasky.di
+
+import android.app.Application
+import android.content.Context
+import android.content.SharedPreferences
+import com.asma.tasky.feature_authentication.data.remote.util.ApiKeyInterceptor
+import com.asma.tasky.core.util.Constants
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object AppModule {
+    @Provides
+    @Singleton
+    fun provideSharedPref(app: Application): SharedPreferences {
+        return app.getSharedPreferences(
+            Constants.SHARED_PREF_NAME,
+            Context.MODE_PRIVATE
+        )
+    }
+
+    @Provides
+    @Singleton
+    fun provideOkHttpClient(sharedPreferences: SharedPreferences): OkHttpClient {
+        return OkHttpClient.Builder()
+            .addInterceptor(ApiKeyInterceptor())
+            .addInterceptor {
+                val token = sharedPreferences.getString(Constants.KEY_JWT_TOKEN, "")
+                val modifiedRequest = it.request().newBuilder()
+                    .addHeader("Authorization", "Bearer $token")
+                    .build()
+                it.proceed(modifiedRequest)
```

Comment:
Instead of an empty string, I'd assign `null` as default value if the token doesn't exist. Then you could also add a check if it's non-null and only then attach the header (else you'll attach an empty Bearer header)